at_lookup/lib/src/connection/base_connection.dart

```diff
@@ -5,12 +5,14 @@ import 'package:at_lookup/src/connection/at_connection.dart';
 
 /// Base class for common socket operations
 abstract class BaseConnection extends AtConnection {
-  final Socket _socket;
+  late final Socket _socket;
   StringBuffer? buffer;
   AtConnectionMetaData? metaData;
 
-  BaseConnection(this._socket) {
+  BaseConnection(Socket? socket) {
     buffer = StringBuffer();
+    socket?.setOption(SocketOption.tcpNoDelay, true);
```

Comment:
Pls, check if we are setting this before the connect. Setting it after connect wont be effective.